lib/fizzy/instantiate.cpp

```diff
@@ -504,7 +507,8 @@ std::optional<ExternalFunction> find_exported_function(Instance& instance, std::
         ExecuteFunction(instance, idx), instance.module->get_function_type(idx)};
```

Comment:
Here we use `*opt_index`. It is weird that `operator*` is not marked noexcept, but docs say "throws nothing". Compared to that `->value()` explicitly checks and throws:

- https://en.cppreference.com/w/cpp/utility/optional/operator*
- https://en.cppreference.com/w/cpp/utility/optional/value

Reply:
I found some explanation, but it's not clear to me https://stackoverflow.com/a/63913833

Reply:
C++ is weird.